docs/user-guide/setting-up-ci-cd.rst

```diff
@@ -0,0 +1,212 @@
+frequently, and automatically testing changes as they are committed. Continuous
+Deployment (CD) means automatically deploying integrated changes to a staging
+or production environment. Together, they allow for more frequent improvements
+to your assistant with less manual effort expended on testing and deployment.
+
+This guide will cover **what** should go in a CI/CD pipeline, specific to a
+Rasa project. **How** you implement that pipeline is up to you. There are many
+CI/CD tools out there, and you might already have a preferred one. Many Git repository 
+hosting services like Github, Gitlab, and Bitbucket provide
+their own CI/CD tools. It is recommended to incorporate
+your CI/CD pipeline into a Git repository set up, instead of using a seperate service.
+
+
+
+Continuous Integration
+----------------------
+
+Assistants are best improved with frequent, `incremental updates
+<https://rasa.com/docs/rasa-x/user-guide/improve-assistant/>`_.
+No matter how small a change is, you want to be sure that it doesn't introduce
+new problems or negatively impact the performance of your assistant. Most tests are
+quick enough to run on every change. However, you can set some more
+resource-intentsive tests to run only when certain files have been changed or
+when a certain tag is present.
+
+CI checks usually run on commit, or on merge/pull request.
+
+.. contents::
+   :local:
+
+Validate Data
+#############
+
+:ref:`Data validation <validate-files>` verifies that there are no mistakes or
+major inconsistencies in your domain file, NLU data, or story data. 
+
+.. code-block:: bash
+
+   rasa data validate --fail-on-warnings
+
+If ``rasa
+data validate`` results in errors, training a model will also fail. By
+including the ``--fail-on-warnings`` flag, the command will also fail on
+warnings about problems that won't prevent training a model, but might indicate
+messy data, such as actions listed in the domain that aren't used in any
+stories.
+
+Validate Stories
+################
+
+:ref:`Story validation <test-story-files-for-conflicts>` checks if you have any
+stories where different bot actions follow from the same dialogue history.
+
+.. code-block:: bash
+
+   rasa data validate --fail-on-warnings --max-history 5
+
+Conflicts between stories will prevent Rasa from learning the correct pattern.
+You can run story validation by passing the ``--max-history`` flag to ``rasa
+data validate``, either in a seperate check or as part of the data validation
+check. ``--max-history`` should be set to the value of ``max_history`` in your
+config for your memoization policy (which defaults to ``5``).
+
+Train a Model
+#############
+
+.. code-block:: bash
+
+   rasa train
+
+Training a model verifies that your NLU pipeline and policy configurations are
+valid and trainable, and it provides a model to use for test conversations. Training a model is also :ref:`part of the continuous deployment
```

Comment:
Maybe rephrase this as "If your model passes tests, then you can use also use this trained model as part of your continuous deployment pipeline"